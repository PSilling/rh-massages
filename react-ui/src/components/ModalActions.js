// react imports
import React, { Component } from 'react';

// util imports
import _t from '../utils/Translations.js';

/**
 * Custom component rendering action buttons for modal footers.
 */
class ModalActions extends Component {

  render() {
    return(
      <div className="pull-right">
        <button type="button" className="btn btn-primary" onClick={this.props.onProceed}
          style={{ 'marginRight': '5px' }} autoFocus={this.props.autoFocus}>
          {this.props.primaryLabel}
        </button>
        <button type="button" className="btn btn-default"
          onClick={this.props.onClose}>
          { _t.translate('Dismiss') }
        </button>
      </div>
    );
  }
}

export default ModalActions
