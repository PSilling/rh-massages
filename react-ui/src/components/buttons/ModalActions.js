// react imports
import React from "react";
import PropTypes from "prop-types";

// util imports
import _t from "../../util/Translations";

/**
 * Footer buttons for Modal dialogs. Children components can be given to add more
 * case specific elements.
 */
const ModalActions = function ModalActions(props) {
  return (
    <div className="pull-right">
      {props.children}
      {props.primaryLabel === "none" ? (
        ""
      ) : (
        <button
          type="button"
          className="btn btn-primary"
          onClick={props.onProceed}
          title={props.title}
          style={{ marginRight: "5px" }}
          autoFocus={props.autoFocus}
          disabled={props.disabled}
        >
          {props.primaryLabel}
        </button>
      )}
      <button type="button" className="btn btn-default" onClick={props.onClose}>
        {_t.translate("Dismiss")}
      </button>
    </div>
  );
};

ModalActions.propTypes = {
  /** callback function triggered on close button click */
  onClose: PropTypes.func.isRequired,
  /** callback function triggered on primary button click */
  onProceed: PropTypes.func.isRequired,
  /** whether the primary button should be automatically focused */
  autoFocus: PropTypes.bool,
  /** extra elements to be provided before the two buttons */
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]),
  /** whether the primary button should be disabled */
  disabled: PropTypes.bool,
  /** primary button label to be displayed */
  primaryLabel: PropTypes.string,
  /** primary button title */
  title: PropTypes.string
};

ModalActions.defaultProps = {
  autoFocus: false,
  children: null,
  disabled: false,
  primaryLabel: _t.translate("Proceed"),
  title: ""
};

export default ModalActions;
