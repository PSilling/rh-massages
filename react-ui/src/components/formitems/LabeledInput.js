// react imports
import React, { Component } from "react";
import PropTypes from "prop-types";

// module imports
import { Col, FormGroup, Label, Input, Tooltip } from "reactstrap";

// util imports
import Util from "../../util/Util";

/**
 * A combined formgroup component with a label (can have a tooltip) and a input.
 * A column is included to set input size.
 */
class LabeledInput extends Component {
  state = { tooltipActive: false };

  tooltipTarget = Util.getTooltipTargets(1)[0];

  toggleTooltip = () => {
    this.setState(prevState => ({ tooltipActive: !prevState.tooltipActive }));
  };

  handleKeyPress = event => {
    if (event.key === "Enter") {
      this.props.onEnterPress();
    }
  };

  render() {
    const { onChange, label, value, onEnterPress, options, size, tooltip, ...rest } = this.props;
    return (
      <Col md={this.props.size}>
        <FormGroup>
          <Label id={this.tooltipTarget} for={`${this.tooltipTarget}_input`}>
            {this.props.label}
          </Label>
          <Input
            {...rest}
            id={`${this.tooltipTarget}_input`}
            value={this.props.value}
            onChange={this.props.onChange}
            onKeyPress={this.handleKeyPress}
            placeholder={this.props.label}
            list={`${this.tooltipTargets}_list`}
          />
          {this.props.options.length > 0 && (
            <datalist id={`${this.tooltipTargets}_list`}>
              {this.props.options.map(item => (
                <option key={item} value={item} />
              ))}
            </datalist>
          )}
          {this.props.tooltip !== "" && (
            <Tooltip isOpen={this.state.tooltipActive} target={this.tooltipTarget} toggle={this.toggleTooltip}>
              {this.props.tooltip}
            </Tooltip>
          )}
        </FormGroup>
      </Col>
    );
  }
}

LabeledInput.propTypes = {
  /** function called on input value change */
  onChange: PropTypes.func.isRequired,
  /** input label text */
  label: PropTypes.string.isRequired,
  /** current value of the input field */
  value: PropTypes.string.isRequired,
  /** input datalist options */
  onEnterPress: PropTypes.func,
  /** input datalist options */
  options: PropTypes.arrayOf(PropTypes.string),
  /** input column length */
  size: PropTypes.string,
  /** input label tooltip */
  tooltip: PropTypes.string
};

LabeledInput.defaultProps = {
  onEnterPress() {},
  options: [],
  size: "12",
  tooltip: ""
};

export default LabeledInput;
