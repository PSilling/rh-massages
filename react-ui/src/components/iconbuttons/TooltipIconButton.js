// react imports
import React, { Component } from "react";
import PropTypes from "prop-types";

// module imports
import { Button, Tooltip } from "reactstrap";

// util imports
import Util from "../../util/Util";

/**
 * Labeless icon only button with an attached tooltip.
 */
class TooltipIconButton extends Component {
  state = { tooltipActive: false };

  tooltipTarget = Util.getTooltipTargets(1)[0];

  toggleTooltip = () => {
    this.setState(prevState => ({ tooltipActive: !prevState.tooltipActive }));
  };

  render() {
    const { disabled, icon, ...rest } = this.props;
    return (
      <span className="no-print">
        <Button {...rest} id={this.tooltipTarget} outline disabled={this.props.disabled}>
          <span className={`fas fa-${this.props.icon}`} />
        </Button>
        {this.props.tooltip !== "" && !this.props.disabled && (
          <Tooltip
            isOpen={this.state.tooltipActive}
            target={this.tooltipTarget}
            toggle={this.toggleTooltip}
            trigger="hover"
          >
            {this.props.tooltip}
          </Tooltip>
        )}
      </span>
    );
  }
}

TooltipIconButton.propTypes = {
  /** icon to be shown instead of a normal label */
  icon: PropTypes.string.isRequired,
  /** function to be called on button click */
  onClick: PropTypes.func.isRequired,
  /** whether the button should be disabled */
  disabled: PropTypes.bool,
  /** button size class */
  size: PropTypes.string,
  /** button component style */
  style: PropTypes.shape({
    border: PropTypes.string
  }),
  /** tooltip to be shown on button hover */
  tooltip: PropTypes.string
};

TooltipIconButton.defaultProps = {
  disabled: false,
  size: "sm",
  style: { border: "0px solid transparent" },
  tooltip: ""
};

export default TooltipIconButton;
