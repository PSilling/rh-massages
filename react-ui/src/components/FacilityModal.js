// react imports
import React, { Component } from 'react';


// component imports
import AddButton from './AddButton';
import ModalActions from './ModalActions';

// module imports
import {ModalContainer, ModalDialog} from 'react-modal-dialog';

// util imports
import _t from '../utils/Translations.js';
import Util from '../utils/Util.js';

class FacilityModal extends Component {

  state = {name: ""}

  /**
   * Sets default input values on props change.
   */
  componentWillReceiveProps(nextProps) {
    if(this.props === nextProps) return;

    this.setState({
      name: (nextProps.facility === -1) ? "" : nextProps.facility.name
    });
  }

  changeName = (event) => {
    this.setState({name: event.target.value});
  }

  /**
   * Handles the post request.
   */
  addFacility = () => {
    Util.post(Util.FACILITIES_URL, {
      name: this.state.name
    }, () => {
      this.props.onToggle();
      this.props.getCallback();
    });
  }

  /**
   * Handles the put request.
   */
  editFacility = () => {
    Util.put(Util.FACILITIES_URL + this.props.facility.id, {
      name: this.state.name
    }, () => {
      this.props.onToggle();
      this.props.getCallback();
    });
  }

  render() {
    return(
      <div className='pull-right'>
        <AddButton onAdd={this.props.onToggle} />

        {this.props.active ?
          <ModalContainer onClose={this.props.onToggle}>
            <ModalDialog onClose={this.props.onToggle} width="50%">
              <h2>
                {this.props.facility === -1 ?
                  _t.translate('New Facility') : _t.translate('Edit Facility')
                }
              </h2>
              <hr />
              <div className="form-group">
                <label>{ _t.translate('Name') }</label>
                <input value={this.state.name} onChange={this.changeName}
                  className="form-control" />
              </div>
              {this.props.facility === -1 ?
                <ModalActions
                  primaryLabel={ _t.translate('Add') }
                  onProceed={this.addFacility}
                  onClose={this.props.onToggle}
                /> :
                <ModalActions
                  primaryLabel={ _t.translate('Edit') }
                  onProceed={this.editFacility}
                  onClose={this.props.onToggle}
                />
              }
            </ModalDialog>
          </ModalContainer> : ''
        }
      </div>
    )
  }
}

export default FacilityModal
