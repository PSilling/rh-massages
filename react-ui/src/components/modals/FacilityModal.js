// react imports
import React, { Component } from "react";
import PropTypes from "prop-types";

// module imports
import { Row, Col, Modal, ModalBody } from "reactstrap";

// component imports
import ModalActions from "../buttons/ModalActions";
import LabeledInput from "../formitems/LabeledInput";
import TooltipIconButton from "../iconbuttons/TooltipIconButton";

// util imports
import _t from "../../util/Translations";
import Fetch from "../../util/Fetch";
import Util from "../../util/Util";

/**
 * Input Modal for Facility management. Allows the change of Facility name.
 * Based on given values can be used for both creating and editing of Facilities.
 */
class FacilityModal extends Component {
  state = { name: "" };

  componentWillReceiveProps(nextProps) {
    if (this.props === nextProps) return;

    this.setState({
      name: nextProps.facility === null ? "" : nextProps.facility.name
    });
  }

  changeName = event => {
    this.setState({ name: event.target.value });
  };

  addFacility = () => {
    if (Util.isEmpty(this.state.name)) {
      Util.notify("error", "", _t.translate("Name is required!"));
      return;
    }
    Fetch.post(
      Util.FACILITIES_URL,
      {
        name: this.state.name
      },
      () => {
        this.props.onToggle();
        this.props.getCallback();
      }
    );
  };

  editFacility = () => {
    if (Util.isEmpty(this.state.name)) {
      Util.notify("error", "", _t.translate("Name is required!"));
      return;
    }
    Fetch.put(
      Util.FACILITIES_URL + this.props.facility.id,
      {
        name: this.state.name
      },
      () => {
        this.props.onToggle();
        this.props.getCallback();
      }
    );
  };

  handleKeyPress = event => {
    if (event.key === "Enter" && document.activeElement.tabIndex === -1) {
      if (this.props.facility === null) {
        this.addFacility();
      } else {
        this.editFacility();
      }
    }
  };

  createInsides = () => (
    <ModalBody>
      <Row>
        <Col md="12">
          <h3>{this.props.facility === null ? _t.translate("New facility") : _t.translate("Edit facility")}</h3>
          <hr />
        </Col>
      </Row>

      <Row>
        <LabeledInput
          label={_t.translate("Name")}
          value={this.state.name}
          onChange={this.changeName}
          onEnterPress={this.props.facility === null ? this.addFacility : this.editFacility}
          type="text"
          maxLength="64"
        />
      </Row>
      {this.props.facility === null ? (
        <ModalActions primaryLabel={_t.translate("Add")} onProceed={this.addFacility} onClose={this.props.onToggle} />
      ) : (
        <ModalActions primaryLabel={_t.translate("Edit")} onProceed={this.editFacility} onClose={this.props.onToggle} />
      )}
    </ModalBody>
  );

  createModal = () =>
    this.props.withPortal ? (
      <Modal size="lg" isOpen toggle={this.props.onToggle} tabIndex="-1" onKeyPress={this.handleKeyPress}>
        {this.createInsides()}
      </Modal>
    ) : (
      this.createInsides()
    );

  render() {
    return (
      <div className="float-right">
        <TooltipIconButton icon="plus" onClick={this.props.onToggle} tooltip={_t.translate("Create a new facility")} />

        {this.props.active && this.createModal()}
      </div>
    );
  }
}

FacilityModal.propTypes = {
  /** whether the dialog should be shown */
  active: PropTypes.bool.isRequired,
  /** callback function for Facility list update */
  getCallback: PropTypes.func.isRequired,
  /** function called on modal toggle */
  onToggle: PropTypes.func.isRequired,
  /** Facility to be possibly edited or null when adding */
  facility: PropTypes.shape({
    id: PropTypes.number,
    name: PropTypes.string
  }),
  /** whether ModalContainer should be used; useful for testing to avoid portals */
  withPortal: PropTypes.bool
};

FacilityModal.defaultProps = {
  facility: null,
  withPortal: true
};

export default FacilityModal;
