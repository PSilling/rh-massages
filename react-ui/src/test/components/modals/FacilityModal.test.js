// react imports
import React from "react";
import { shallow } from "enzyme";

// test imports
import FacilityModal from "../../../components/modals/FacilityModal";
import LabeledInput from "../../../components/formitems/LabeledInput";
import ModalActions from "../../../components/buttons/ModalActions";
import TooltipIconButton from "../../../components/iconbuttons/TooltipIconButton";
import _t from "../../../util/Translations";

// test mocks
jest.mock("../../../util/Fetch");

test("renders inside content with correct props", () => {
  const testGetFunction = jest.fn();
  const testToggleFunction = jest.fn();
  const wrapper = shallow(
    <FacilityModal
      active
      facility={null}
      getCallback={testGetFunction}
      onToggle={testToggleFunction}
      withPortal={false}
    />
  );
  const button = wrapper.find(TooltipIconButton);
  const actions = wrapper.find(ModalActions);
  const heading = wrapper.find("h3");
  const input = wrapper.find(LabeledInput);

  expect(testToggleFunction).not.toHaveBeenCalled();

  actions.props().onClose();

  expect(testToggleFunction).toHaveBeenCalledTimes(1);
  expect(testGetFunction).not.toHaveBeenCalled();

  actions.props().onProceed();

  expect(testToggleFunction).toHaveBeenCalledTimes(1);
  expect(testGetFunction).not.toHaveBeenCalled();
  expect(button.props().onClick).toBe(testToggleFunction);
  expect(heading.props().children).toEqual(_t.translate("New facility"));
  expect(input.props().value).toBe("");
  expect(wrapper).toMatchSnapshot();
});

test("switches to edit mode when a Facility is given", () => {
  const testGetFunction = jest.fn();
  const testToggleFunction = jest.fn();
  const testFacility = { id: 1, name: "test" };
  const wrapper = shallow(
    <FacilityModal
      active
      facility={testFacility}
      getCallback={testGetFunction}
      onToggle={testToggleFunction}
      withPortal={false}
    />
  );
  wrapper.instance().setState({ name: testFacility.name, facility: testFacility });

  const actions = wrapper.find(ModalActions);
  const heading = wrapper.find("h3");
  const input = wrapper.find(LabeledInput);

  expect(testToggleFunction).not.toHaveBeenCalled();
  expect(testGetFunction).not.toHaveBeenCalled();

  actions.props().onProceed();

  expect(testToggleFunction).toHaveBeenCalledTimes(1);
  expect(testGetFunction).toHaveBeenCalledTimes(1);
  expect(heading.props().children).toEqual(_t.translate("Edit facility"));
  expect(actions.props().primaryLabel).toBe(_t.translate("Edit"));
  expect(input.props().value).toBe("test");
  jest.resetAllMocks();
});
