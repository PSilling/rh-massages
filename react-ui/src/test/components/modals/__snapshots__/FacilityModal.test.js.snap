// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders inside content with correct props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FacilityModal
    active={true}
    facility={null}
    getCallback={[MockFunction]}
    onToggle={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    withPortal={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <TooltipIconButton
          disabled={false}
          icon="plus"
          onClick={
            [MockFunction] {
              "calls": Array [
                Array [],
              ],
              "results": Array [
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
              ],
            }
          }
          size="sm"
          style={
            Object {
              "border": "0px solid transparent",
            }
          }
          tooltip="Vytvořit novou budovu"
        />,
        <ModalBody
          tag="div"
        >
          <Row
            tag="div"
          >
            <Col
              md="12"
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              <h3>
                Nová budova
              </h3>
              <hr />
            </Col>
          </Row>
          <Row
            tag="div"
          >
            <LabeledInput
              label="Název"
              maxLength="64"
              onChange={[Function]}
              onEnterPress={[Function]}
              options={Array []}
              size="12"
              tooltip=""
              type="text"
              value=""
            />
          </Row>
          <ModalActions
            disabled={false}
            onClose={
              [MockFunction] {
                "calls": Array [
                  Array [],
                ],
                "results": Array [
                  Object {
                    "isThrow": false,
                    "value": undefined,
                  },
                ],
              }
            }
            onProceed={[Function]}
            primaryLabel="Přidat"
          />
        </ModalBody>,
      ],
      "className": "float-right",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "disabled": false,
          "icon": "plus",
          "onClick": [MockFunction] {
            "calls": Array [
              Array [],
            ],
            "results": Array [
              Object {
                "isThrow": false,
                "value": undefined,
              },
            ],
          },
          "size": "sm",
          "style": Object {
            "border": "0px solid transparent",
          },
          "tooltip": "Vytvořit novou budovu",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Row
              tag="div"
            >
              <Col
                md="12"
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <h3>
                  Nová budova
                </h3>
                <hr />
              </Col>
            </Row>,
            <Row
              tag="div"
            >
              <LabeledInput
                label="Název"
                maxLength="64"
                onChange={[Function]}
                onEnterPress={[Function]}
                options={Array []}
                size="12"
                tooltip=""
                type="text"
                value=""
              />
            </Row>,
            <ModalActions
              disabled={false}
              onClose={
                [MockFunction] {
                  "calls": Array [
                    Array [],
                  ],
                  "results": Array [
                    Object {
                      "isThrow": false,
                      "value": undefined,
                    },
                  ],
                }
              }
              onProceed={[Function]}
              primaryLabel="Přidat"
            />,
          ],
          "tag": "div",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Col
                md="12"
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <h3>
                  Nová budova
                </h3>
                <hr />
              </Col>,
              "tag": "div",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <h3>
                    Nová budova
                  </h3>,
                  <hr />,
                ],
                "md": "12",
                "tag": "div",
                "widths": Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Nová budova",
                  },
                  "ref": null,
                  "rendered": "Nová budova",
                  "type": "h3",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "hr",
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <LabeledInput
                label="Název"
                maxLength="64"
                onChange={[Function]}
                onEnterPress={[Function]}
                options={Array []}
                size="12"
                tooltip=""
                type="text"
                value=""
              />,
              "tag": "div",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "label": "Název",
                "maxLength": "64",
                "onChange": [Function],
                "onEnterPress": [Function],
                "options": Array [],
                "size": "12",
                "tooltip": "",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": null,
              "disabled": false,
              "onClose": [MockFunction] {
                "calls": Array [
                  Array [],
                ],
                "results": Array [
                  Object {
                    "isThrow": false,
                    "value": undefined,
                  },
                ],
              },
              "onProceed": [Function],
              "primaryLabel": "Přidat",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <TooltipIconButton
            disabled={false}
            icon="plus"
            onClick={
              [MockFunction] {
                "calls": Array [
                  Array [],
                ],
                "results": Array [
                  Object {
                    "isThrow": false,
                    "value": undefined,
                  },
                ],
              }
            }
            size="sm"
            style={
              Object {
                "border": "0px solid transparent",
              }
            }
            tooltip="Vytvořit novou budovu"
          />,
          <ModalBody
            tag="div"
          >
            <Row
              tag="div"
            >
              <Col
                md="12"
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <h3>
                  Nová budova
                </h3>
                <hr />
              </Col>
            </Row>
            <Row
              tag="div"
            >
              <LabeledInput
                label="Název"
                maxLength="64"
                onChange={[Function]}
                onEnterPress={[Function]}
                options={Array []}
                size="12"
                tooltip=""
                type="text"
                value=""
              />
            </Row>
            <ModalActions
              disabled={false}
              onClose={
                [MockFunction] {
                  "calls": Array [
                    Array [],
                  ],
                  "results": Array [
                    Object {
                      "isThrow": false,
                      "value": undefined,
                    },
                  ],
                }
              }
              onProceed={[Function]}
              primaryLabel="Přidat"
            />
          </ModalBody>,
        ],
        "className": "float-right",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "disabled": false,
            "icon": "plus",
            "onClick": [MockFunction] {
              "calls": Array [
                Array [],
              ],
              "results": Array [
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
              ],
            },
            "size": "sm",
            "style": Object {
              "border": "0px solid transparent",
            },
            "tooltip": "Vytvořit novou budovu",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <Row
                tag="div"
              >
                <Col
                  md="12"
                  tag="div"
                  widths={
                    Array [
                      "xs",
                      "sm",
                      "md",
                      "lg",
                      "xl",
                    ]
                  }
                >
                  <h3>
                    Nová budova
                  </h3>
                  <hr />
                </Col>
              </Row>,
              <Row
                tag="div"
              >
                <LabeledInput
                  label="Název"
                  maxLength="64"
                  onChange={[Function]}
                  onEnterPress={[Function]}
                  options={Array []}
                  size="12"
                  tooltip=""
                  type="text"
                  value=""
                />
              </Row>,
              <ModalActions
                disabled={false}
                onClose={
                  [MockFunction] {
                    "calls": Array [
                      Array [],
                    ],
                    "results": Array [
                      Object {
                        "isThrow": false,
                        "value": undefined,
                      },
                    ],
                  }
                }
                onProceed={[Function]}
                primaryLabel="Přidat"
              />,
            ],
            "tag": "div",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <Col
                  md="12"
                  tag="div"
                  widths={
                    Array [
                      "xs",
                      "sm",
                      "md",
                      "lg",
                      "xl",
                    ]
                  }
                >
                  <h3>
                    Nová budova
                  </h3>
                  <hr />
                </Col>,
                "tag": "div",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": Array [
                    <h3>
                      Nová budova
                    </h3>,
                    <hr />,
                  ],
                  "md": "12",
                  "tag": "div",
                  "widths": Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Nová budova",
                    },
                    "ref": null,
                    "rendered": "Nová budova",
                    "type": "h3",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": "hr",
                  },
                ],
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <LabeledInput
                  label="Název"
                  maxLength="64"
                  onChange={[Function]}
                  onEnterPress={[Function]}
                  options={Array []}
                  size="12"
                  tooltip=""
                  type="text"
                  value=""
                />,
                "tag": "div",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "label": "Název",
                  "maxLength": "64",
                  "onChange": [Function],
                  "onEnterPress": [Function],
                  "options": Array [],
                  "size": "12",
                  "tooltip": "",
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": null,
                "disabled": false,
                "onClose": [MockFunction] {
                  "calls": Array [
                    Array [],
                  ],
                  "results": Array [
                    Object {
                      "isThrow": false,
                      "value": undefined,
                    },
                  ],
                },
                "onProceed": [Function],
                "primaryLabel": "Přidat",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
